app/controllers/lead_providers/confirm_schools_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class ConfirmSchoolsController < ApplicationController
+    before_action :load_form
+
+    def show
+      @schools = School.where(id: @form.school_ids).order(:name)
+      @delivery_partner = DeliveryPartner.find(@form.delivery_partner_id)
+    end
+
+    def remove
+      school = School.find @form.school_ids.delete(params[:remove][:school_id])
```

Comment:
I got lost somewhere between third and fourth thing that this line is doing :P 
Can we split it into a few? Maybe one to get the id, another one to find school, another one to remove the school id from the form?